wagtail/admin/rich_text/converters/html_to_contentstate.py

```diff
@@ -206,12 +206,19 @@ def get_attribute_data(self, attrs):
         try:
             page = Page.objects.get(id=attrs['id']).specific
         except Page.DoesNotExist:
-            return {}
+            # retain ID so that it's still identified as a page link (albeit a broken one)
+            return {
+                'id': int(attrs['id']),
+                'url': None,
+                'parentId': None
+            }
+
+        parent_page = page.get_parent()
 
         return {
             'id': page.id,
             'url': page.url,
-            'parentId': page.get_parent().id,
+            'parentId': parent_page.id if parent_page else None,
```

Comment:
Under what circumstances would a page exist and not have a parent page?

Reply:
@thibaudcolas When it's the root page. We don't allow the root page to be picked as of dc093c96461ab1e40fe0b42bc7ee09383bf8d71d, but this way we avoid breaking on links that already existed prior to this fix (and it's more robust to have a server-side check for this case).